rust/src/node_manager/cached_node_manager.rs

```diff
@@ -357,6 +364,56 @@ mod tests {
         manager.checkpoint().expect("checkpoint should succeed");
     }
 
+    #[test]
+    fn cached_node_manager_restore_calls_restore_on_underlying_storage() {
+        static RESTORE_CALLED: AtomicBool = AtomicBool::new(false);
+
+        struct CheckRestoreStorage;
+
+        impl Storage for CheckRestoreStorage {
+            type Id = u32;
+            type Item = i32;
+
+            fn open(_path: &Path) -> BTResult<Self, storage::Error> {
+                unimplemented!()
+            }
+
+            fn get(&self, _id: Self::Id) -> BTResult<Self::Item, storage::Error> {
+                unimplemented!()
+            }
+
+            fn reserve(&self, _item: &Self::Item) -> Self::Id {
+                unimplemented!()
+            }
+
+            fn set(&self, _id: Self::Id, _item: &Self::Item) -> BTResult<(), storage::Error> {
+                unimplemented!()
+            }
+
+            fn delete(&self, _id: Self::Id) -> BTResult<(), storage::Error> {
+                unimplemented!()
+            }
+
+            fn close(self) -> BTResult<(), storage::Error> {
+                unimplemented!()
+            }
+        }
+
+        impl Checkpointable for CheckRestoreStorage {
+            fn checkpoint(&self) -> BTResult<u64, crate::storage::Error> {
+                unimplemented!()
+            }
+
+            fn restore(_path: &Path, _checkpoint: u64) -> BTResult<(), crate::storage::Error> {
+                RESTORE_CALLED.store(true, Ordering::Relaxed);
+                Ok(())
+            }
+        }
+
+        CachedNodeManager::<CheckRestoreStorage>::restore(Path::new("/some/path"), 1).unwrap();
+        assert!(RESTORE_CALLED.load(Ordering::Relaxed));
```

Comment:
```suggestion
        let ctx = MockCachedNodeManagerStorage::restore_context();
        ctx.expect()
            .times(1)
            .with(eq(Path::new("/some/path")), eq(1))
            .returning(|_, _| Ok(()));

        CachedNodeManager::<MockCachedNodeManagerStorage>::restore(Path::new("/some/path"), 1)
            .unwrap();
```
It's the only test that calls `restore`, I think it's fine to use static mock here